src/dftbp/dftbplus/main.F90

```diff
@@ -2881,27 +3088,32 @@ subroutine buildAndDiagDenseRealHam(env, denseDesc, ints, species, neighbourList
     !> species of all atoms in the system
     integer, intent(in) :: species(:)
 
+    !> Is the system periodic?
+    logical, intent(in) :: tPeriodic
+
     !> Is the geometry helical
     logical, intent(in) :: tHelical
 
     !> Coordinates of all atoms including images
-    real(dp), allocatable, intent(inout) :: coord(:,:)
+    real(dp), intent(inout), allocatable :: coord(:,:)
 
     !> Electronic solver information
     type(TElectronicSolver), intent(inout) :: electronicSolver
 
     !> K-points and spins to be handled
     type(TParallelKS), intent(in) :: parallelKS
 
-    !>Data for rangeseparated calculation
-    type(TRangeSepFunc), allocatable, intent(inout) :: rangeSep
+    !>Data for hybrid xc-functional calculation
+    class(THybridXcFunc), intent(inout), allocatable :: hybridXc
 
-    !> Change in density matrix during last rangesep SCC cycle
-    real(dp), pointer, intent(in) :: deltaRhoInSqr(:,:,:)
+    !> Change in density matrix during last hybridXc SCC cycle
+    real(dp), intent(in) :: deltaRhoIn(:,:,:)
```

Comment:
As this won't be allocated for non-hybrid calculations
```suggestion
    real(dp), intent(in), allocatable :: deltaRhoIn(:,:,:)
```

Reply:
Thanks for pointing this out; fixed by now.